File: C++/13. Palindrome Check/Palindrome Check [O(n)].cpp
#include <iostream>
#include <vector>
#include <numeric>

using namespace std;

// O(n) time | O(n) space

bool isPalindrome(string str) {
	vector<char> reversedChars;
	for (int i = str.length() - 1; i >= 0; i--) {
		reversedChars.push_back(str[i]);
	}
	return str == string(reversedChars.begin(), reversedChars.end());
}

File: AlgoExpert/1. Easy/Java/13. Palindrome Check/Palindrome Check [O(n)].java
class Program {

	// O(n) time | O(n) space

	public static boolean isPalindrome(String str) {
		StringBuilder reversedString = new StringBuilder();
		for (int i = str.length() - 1; i >= 0; i--) {
			reversedString.append(str.charAt(i));
		}
		return str.equals(reversedString.toString());
	}
}